src/execution_plan/optimizations/seek_by_id.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "seek_by_id.h"
+#include "../../util/arr.h"
+#include "../ops/op_filter.h"
+#include "../ops/op_index_scan.h"
+#include "../ops/op_all_node_scan.h"
+#include "../ops/op_node_by_id_seek.h"
+#include "../ops/op_node_by_label_scan.h"
+
+static bool _idFilter(FT_FilterNode *f, int *rel, EntityID *id) {
+    if(f->t == FT_N_COND) return false;
+    if(f->pred.op == NE) return false;
+    
+    AR_OpNode *op;
+    AR_OperandNode *operand;
+    AR_ExpNode *lhs = f->pred.lhs;
+    AR_ExpNode *rhs = f->pred.rhs;
+    *rel = f->pred.op;
+
+    /* Either ID(N) compare const
+     * OR
+     * const compare ID(N) */
+    if(lhs->type == AR_EXP_OPERAND && rhs->type == AR_EXP_OP) {
+        op = &rhs->op;
+        operand = &lhs->operand;
+    } else if(lhs->type == AR_EXP_OP && rhs->type == AR_EXP_OPERAND) {
+        op = &lhs->op;
+        operand = &rhs->operand;
+    } else {
+        return false;
+    }
+
+    // Make sure ID is compared to a constant.
+    if(operand->type != AR_EXP_CONSTANT) return false;
+    if(SI_TYPE(operand->constant) != T_INT64) return false;
+    *id = SI_GET_NUMERIC(operand->constant);
+
+    // Make sure applied function is ID.
+    if(strcasecmp(op->func_name, "id")) return false;
+
+    return true;
+}
+
+static void _setupIdRange(int rel, EntityID id, NodeID *minId, NodeID *maxId, bool *includeMin, bool *includeMax) {
+    switch(rel) {
+        case GT:
+            *minId = id;
+            break;
+        case GE:
+            *minId = id;
+            *includeMin = true;
+            break;
+        case LT:
+            *maxId = id;
```

Comment:
Technically, cases like `ID(a) < 1` are min-inclusive, `ID(a) >= 1` are max-inclusive, etc.

It seems like the range checkers make this a non-issue, though, so no need for change?

Reply:
renamed to inclusiveMin and inclusiveMax